### ISSUE TITLE
Only detection?

### ISSUE THREAD

**Issue by HanG-CS** [Yuanchu/YOLO3D#7]
Are you only detect the object but not classify it?

**Comment by Artamus** [comment]
I would like to know about this as well, both this and the other ComplexYOLO code seem to not output any class predictions. From the code it seems they are included (`output[7:]` seems to be the magic location), but due to my unfamiliarity with both YOLO and PyTorch I simply cannot crack how I'm supposed to get a meaningful prediction.

**Comment by abhigoku10** [comment]
@Artamus @HanG-CS  i am also facing the same issues . Can you classify the bounding box or obtain positive predictions on the bounding box ?

**Comment by Artamus** [comment]
Yes, I figured it out. Basically the output matrix you get, the last columns are the class scores, i.e.
they are assembled like this: `pred_boxes[..., 7:(7 + nC) ] = pred_cls`.
So the last columns from the method `get_region_boxes(output, conf_thresh, num_classes, anchors, num_anchors)` are what you are looking for.

**Comment by abhigoku10** [comment]
@Artamus  thanks for the pointers, i was able to get the class label . The issue which i am facing now is that their misclassifcation in the class label.? are you also facing the same issue. I have tried to use sigmoid and softmax both are yielding same result

**Comment by Artamus** [comment]
If you just want to get the highest scored class, `argmax` should be enough (i.e. just get the class with highest score, no need to pass it to a softmax). 

As to the accuracy, sorry, I cannot help you there, I only did a quick evaluation of ComplexYOLO and I am not using it anywhere.

**Comment by Hizhaoyuan** [comment]
@Artamus Hi, you can take a look at this project, which has a complete implementation and effect presentation[https://github.com/wwooo/tensorflow_complex_yolo](url)
